.github/workflows/smoke-test.yml

```diff
@@ -1,33 +1,29 @@
 name: Smoke Test
 
 on:
-  pull_request:
-  push:
+  workflow_run:
+    workflows: ["Build"]
+    types:
+      - completed
 
 defaults:
   run:
     shell: 'bash -Eeuo pipefail -x {0}'
 
 jobs:
   smoke-test:
-    name: Snapcraft
     runs-on: ubuntu-22.04
+    if: ${{ github.event.workflow_run.conclusion == 'success' }}
     steps:
-      - uses: actions/checkout@v2
+      - name: Checkout code
+        uses: actions/checkout@v4
 
-      - name: Build
-        # https://github.com/snapcore/action-build/releases
-        uses: snapcore/action-build@v1
+      - name: Get the artifact
+        uses: dawidd6/action-download-artifact@v6
         with:
-          snapcraft-args: snap --output docker.snap
-
-      - name: Upload
-        # TODO the goal of this "if:" condition is to avoid creating artifacts on private forks (and using up all their storage quota), but Tianon couldn't find a clean way to do that so this was the next best thing
-        if: github.repository == 'docker-snap/docker-snap'
-        uses: actions/upload-artifact@v2
-        with:
-          name: docker.snap
-          path: docker.snap
+          workflow: build.yml
```

Comment:
This is actually very fragile. Can it be replaced with the workflow name, matching the name given in the trigger (`"Build"`)?

From the docs:
```
   # Optional, workflow file name or ID
   # If not specified, will be inferred from run_id (if run_id is specified), or will be the current workflow
   workflow: workflow_name.yml
```
Looking at the action's source [here](https://github.com/dawidd6/action-download-artifact/blob/db9477a3ebd885bf3bb8da6a98eeb0d6da2d0d9a/main.js#L60-L66), it seems that by workflow ID, they mean its name.